src/containerapp-preview/azext_containerapp_preview/_utils.py

```diff
@@ -0,0 +1,92 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from azure.cli.core.azclierror import NoTTYError
+from azure.cli.core.extension.dynamic_install import _get_extension_use_dynamic_install_config
+from knack.prompting import prompt_y_n, NoTTYException
+from knack.util import CLIError
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+MIN_GA_VERSION = '0.3.35'
+GA_CONTAINERAPP_EXTENSION_NAME = 'containerapp'
+GA_CONTAINERAPP_EXTENSION_MODULE = 'azext_containerapp.custom'
+
+
+def _get_or_add_extension(cmd, extension_name):
+    from azure.cli.core.extension import (
+        ExtensionNotInstalledException, get_extension)
+    from pkg_resources import parse_version
+    prompt_ext = True
+    try:
+        ext = get_extension(extension_name)
+        # check extension version
+        if ext and parse_version(ext.version) < parse_version(MIN_GA_VERSION):
+            msg = f"The command requires the latest version of extension {extension_name}. Run 'az extension add --upgrade -n {extension_name}' to upgrade extension"
+            logger.Error(msg)
+            return False
```

Comment:
We usually do not recommend relying on one extension to another because it introduces complex version dependency issues.

Reply:
This containerapp-preview extension relies on containerapp extension. Need to prompt the user to upgrade the GA version of the extension in preview extension.